osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmanagement/application/endpoints/SecretManagementEndpoint.java

```diff
@@ -0,0 +1,138 @@
+import org.opensmartgridplatform.shared.exceptionhandling.ComponentType;
+import org.opensmartgridplatform.shared.exceptionhandling.OsgpException;
+import org.opensmartgridplatform.shared.exceptionhandling.TechnicalException;
+import org.opensmartgridplatform.ws.schema.core.secret.management.GetSecretsRequest;
+import org.opensmartgridplatform.ws.schema.core.secret.management.GetSecretsResponse;
+import org.opensmartgridplatform.ws.schema.core.secret.management.OsgpResultType;
+import org.opensmartgridplatform.ws.schema.core.secret.management.SecretTypes;
+import org.opensmartgridplatform.ws.schema.core.secret.management.StoreSecretsRequest;
+import org.opensmartgridplatform.ws.schema.core.secret.management.StoreSecretsResponse;
+import org.opensmartgridplatform.ws.schema.core.secret.management.TypedSecrets;
+import org.springframework.ws.server.endpoint.annotation.Endpoint;
+import org.springframework.ws.server.endpoint.annotation.PayloadRoot;
+import org.springframework.ws.server.endpoint.annotation.RequestPayload;
+import org.springframework.ws.server.endpoint.annotation.ResponsePayload;
+
+@Endpoint
+@Slf4j
+public class SecretManagementEndpoint {
+
+    private static final String NAMESPACE_URI = "http://www.opensmartgridplatform"
+            + ".org/schemas/security/secretmanagement";
+
+    private final SecretManagementService secretManagementService;
+    private final SoapEndpointDataTypeConverter converter;
+
+    public SecretManagementEndpoint(SecretManagementService secretManagementService,
+            SoapEndpointDataTypeConverter converter) {
+        this.secretManagementService = secretManagementService;
+        this.converter = converter;
+    }
+
+    @PayloadRoot(namespace = NAMESPACE_URI, localPart = "getSecretsRequest")
+    @ResponsePayload
+    public GetSecretsResponse getSecretsRequest(@RequestPayload GetSecretsRequest request) throws OsgpException {
+
+        log.info("Handling incoming SOAP request 'getSecretsRequest' for device {}", request.getDeviceId());
+
+        if (log.isDebugEnabled()) {
+            log.debug(getSecretsRequestToString(request));
+        }
+
+        try {
+            GetSecretsResponse response = new GetSecretsResponse();
+
+            SecretTypes soapSecretTypes = request.getSecretTypes();
+
+            if (soapSecretTypes == null) {
+                throw new TechnicalException("Missing input: secret types");
+            }
+
+            List<SecretType> secretTypeList = converter.convertToSecretTypes(soapSecretTypes);
+            List<TypedSecret> typedSecrets = secretManagementService.retrieveSecrets(request.getDeviceId(),
+                    secretTypeList);
+
+            TypedSecrets soapTypedSecrets = converter.convertToSoapTypedSecrets(typedSecrets);
+
+            response.setTypedSecrets(soapTypedSecrets);
+            response.setResult(OsgpResultType.OK);
+
+            log.trace(response.toString());
+
+            return response;
+
+        } catch (Exception e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1166)